docs/reference/annotations.md

```diff
@@ -76,30 +76,73 @@ annotation class Suspendable
 val f = @Suspendable { Fiber.sleep(10) }
 ```
 
+## Annotation Use-site Targets
 
-## Java Annotations
+When you're annotating a property or a primary constructor parameter, there are multiple Java elements which are
+generated from the corresponding Kotlin element, and therefore multiple possible locations for the annotation in
+the generated Java bytecode. To specify how exactly the annotation should be generated, use the following syntax:
 
-Java annotations are 100% compatible with Kotlin:
+``` kotlin
+class Example(@field:Ann val foo,    // annotate Java field
+              @get:Ann val bar,      // annotate Java getter
+              @param:Ann val quux)   // annotate Java constructor parameter
+```
+
+The same syntax can be used to annotate the entire file. To do this, put an annotation with the target `file` at
+the top level of a file:
 
 ``` kotlin
-import org.junit.Test
-import org.junit.Assert.*
+@file:JvmName("Foo")
+```
```

Comment:
I'd clarify: before the package directive, or all import if the package directive is not there